packages/backend/src/modules/tvl2/indexers/ValueIndexer.ts

```diff
@@ -11,137 +9,70 @@ import {
-  canonicalForTotal: bigint
-  external: bigint
-  externalForTotal: bigint
-  native: bigint
-  nativeForTotal: bigint
-}
-
-type AssetId = string
-type PriceId = string
-
-/**
- * Indexer that aggregates TVL for a project. Skips the configurations that are not included in the total.
- */
 export class ValueIndexer extends ManagedChildIndexer {
-  private readonly amountConfigs: (AmountConfigEntry & { configId: string })[]
+  private readonly amountConfigs: Map<AmountId, AmountConfigEntry>
   private readonly priceConfigIds: Map<AssetId, PriceId>
 
   constructor(private readonly $: ValueIndexerDeps) {
     const logger = $.logger.tag($.tag)
     const name = 'value_indexer'
     super({ ...$, name, logger })
 
-    this.amountConfigs = $.amountConfigs.map((x) => ({
-      ...x,
-      configId: createAmountId(x),
-    }))
+    this.amountConfigs = getAmountConfigs($.amountConfigs)
     this.priceConfigIds = getPriceConfigIds($.priceConfigs)
   }
 
   override async update(from: number, to: number): Promise<number> {
-    // Potential future optimization
-    // check if db.configHash === this.configHash
-    // YES - skip update
-    // NO - continue update
-
-    const timestamp = this.$.syncOptimizer.getTimestampToSync(from)
-    if (timestamp.toNumber() > to) {
-      this.logger.info('Skipping update due to sync optimization', {
+    if (this.$.minHeight > to) {
+      this.logger.info('Skipping update due to minHeight', {
         from,
         to,
-        optimizedTimestamp: timestamp.toNumber(),
+        minHeight: this.$.minHeight,
       })
       return to
     }
-    const configIds = this.amountConfigs
-      .filter((x) => x.sinceTimestamp.gte(timestamp))
-      .map((x) => x.configId)
-    if (configIds.length === 0) {
-      this.logger.info('Skipping update due to no configs', {
-        from,
-        to,
-        timestamp: timestamp.toNumber(),
-      })
-      return timestamp.toNumber()
-    }
-
-    const value = await this.getTvlAt(timestamp, configIds)
-    await this.$.valueRepo.addOrUpdate({
-      projectId: this.$.project,
-      timestamp,
-      dataSource: this.$.dataSource,
-      ...value,
-    })
 
-    return timestamp.toNumber()
-  }
-
-  async getTvlAt(timestamp: UnixTime, configIds: string[]): Promise<Values> {
-    const records = await this.$.amountRepo.getByConfigIdsAndTimestamp(
-      configIds,
-      timestamp,
-    )
+    const timestamps: UnixTime[] = []
 
-    const prices = await this.$.priceRepo.getByTimestamp(timestamp)
-
-    const results = {
-      canonical: 0n,
-      canonicalForTotal: 0n,
-      external: 0n,
-      externalForTotal: 0n,
-      native: 0n,
-      nativeForTotal: 0n,
+    let current = from > this.$.minHeight ? from : this.$.minHeight
+    while (
+      this.$.syncOptimizer.getTimestampToSync(current).toNumber() <= to &&
+      timestamps.length < this.$.maxTimestampsToProcessAtOnce
+    ) {
+      timestamps.push(this.$.syncOptimizer.getTimestampToSync(current))
```

Comment:
Please check if there's at least one configuration to sync before adding a timestamp to this array (recent addition to `ValueIndexer`)